### PR TITLE
Request component

### PR DESCRIPTION
**Pull request by lizbanach**
Screen Shot 2020-03-25 at 11 09 36 PM

- request component
- added util files

### CODE REVIEW

File: src/components/Request/Request.styles.js [hunk 1]
```diff
@@ -0,0 +1,44 @@
...
+
+export const styles = {
+  amount: css({
+    color: Color.CORAL,
+    fontWeight: 500,
+    paddingLeft: Space.S5,
+  }),
+  content: css({
+    color: Color.GRAY_50,
+    fontWeight: 500,
```
Comment (izgeorge) [inline src/components/Request/Request.styles.js]: should we add this to `TEXT_TYPE`?

File: src/components/Request/index.js [hunk 2]
```diff
@@ -0,0 +1,66 @@
...
+        </Text>
+      </div>
+      <div css={styles.progressBar}>
+        <div
+          css={styles.progress}
+          css={{
+            background: Color.CORAL,
+            borderRadius: Radius.ROUNDED,
+            height: Space.S10,
+            width: `${progress}%`,
```
Comment (izgeorge) [inline src/components/Request/index.js]: can you add the static attrs to styles.progress and then compose?

`css={[styles.progress, { width: `${progress}`% }]}

File: src/lib/utils/datetime.js [hunk 3]
```diff
@@ -0,0 +1,7 @@
+import moment from 'moment'
+
+// Format incoming date
+export const formatDate = (date) => {
+  const momentDate = moment(date || undefined)
```
Comment (izgeorge) [inline src/lib/utils/datetime.js]: will passing undefined fall back as `now`?

File: src/components/Request/index.js [hunk 4]
```diff
@@ -0,0 +1,66 @@
...
+import { formatDate } from 'lib/utils/datetime'
+import { numberWithCommas } from 'lib/utils/number'
+import { SecondaryButton } from 'components/Button'
+import Text from 'components/Text'
+import { TEXT_TYPE } from 'components/Text/constants'
+import { styles } from './Request.styles'
+
+export const Request = ({ date, donation, id, name, onDelete, request }) => {
+  const numberDonation = numericString(donation)
+  const numberRequest = numericString(request)
```
Comment (izgeorge) [inline src/components/Request/index.js]: this function in the other project was used to strip out non-number characters. do the donations/requests contain anything besides ints?

it looks like it doesn't given that we are running `numberWithCommas` on it. if that's the case, you can use the unary operatory `+` before the number.

File: src/components/Request/index.js (hunk 2 above)
Comment (lizbanach) [reply src/components/Request/index.js]: @izgeorge i tried this initially and it didn't work :(

Comment (izgeorge) [reply src/components/Request/index.js]: 🤔  weird, did you try wrapping it with the css func?

Comment (lizbanach) [reply src/components/Request/index.js]: that's a good idea! let me try that

File: src/components/Request/Request.styles.js (hunk 1 above)
Comment (lizbanach) [reply src/components/Request/Request.styles.js]: @izgeorge good idea

File: src/lib/utils/datetime.js (hunk 3 above)
Comment (lizbanach) [reply src/lib/utils/datetime.js]: @izgeorge it * shouldn't * ever be undefined, but yes we should have a fallback to now if its undefined

File: src/components/Request/index.js (hunk 2 above)
Comment (lizbanach) [reply src/components/Request/index.js]: sike! that worked! thank you <3

**Pull request merged**